lib/Jsrt/ChakraDebug.h

```diff
@@ -0,0 +1,567 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+/// \mainpage Chakra Hosting Debugging API Reference
+///
+/// Chakra is Microsoft's JavaScript engine. It is an integral part of Internet Explorer but can
+/// also be hosted independently by other applications. This reference describes the APIs available
+/// to applications to debug JavaScript.
+
+/// \file
+/// \brief The Chakra hosting debugging API.
+///
+/// This file contains a flat C API layer. This is the API exported by ChakraCore.dll.
+
+#ifdef _MSC_VER
+#pragma once
+#endif  // _MSC_VER
+
+#ifndef _CHAKRADEBUG_H_
+#define _CHAKRADEBUG_H_
+
+#include <windows.h>
+
+#include "ChakraCommon.h"
+
+    /// <summary>
+    ///     Debug events reported from ChakraCore engine.
+    /// </summary>
+    typedef enum _JsDiagDebugEvent
+    {
+        /// <summary>
+        ///     Indicates a new script being compiled, this includes script, eval, new function.
+        /// </summary>
+        JsDiagDebugEventSourceCompile= 0,
+        /// <summary>
+        ///     Indicates compile error for a script.
+        /// </summary>
+        JsDiagDebugEventCompileError = 1,
+        /// <summary>
+        ///     Indicates a break due to a breakpoint.
+        /// </summary>
+        JsDiagDebugEventBreak = 2,
```

Comment:
If this is exclusively break point, why not rename this to `JsDiagDebugEventBreakPoint`? Would be clearer this way since the enum values below this are technically breaks as well.